### ISSUE TITLE
[BUG]: Modal Route does not support navigation with browser history

### ISSUE THREAD

**Issue by enkelmedia** [umbraco/Umbraco-CMS#16684]
I just notice that the behavior of the routed modal (`UmbModalRouteRegistrationController`) is quite strange when clicking on the "back"-button in the browser.

The route changes in the address bar and the view "behind" the modal is changing - but the modal is still on top of everything.

![umb-modal-strange](https://github.com/umbraco/Umbraco.CMS.Backoffice/assets/1782524/92619ff4-aea9-4aae-bf8e-fa46bebdc0c1)

In an ideal world there should be some kind of callback, like the `onSetup()`, `onReject()` to hook into the navigation and maybe prompt the user if "the thing" in the modal is not saved.

I played with some of the events from the `router-slot` library but the route still changed.

```typescript
window.addEventListener("willchangestate", e => {

    // Check if we should navigate away from this page
    if (!confirm("You have unsaved data. Do you wish to discard it?")) {
        e.preventDefault();
        return;
    }

}, {once: true});
```

**Comment by nielslyngsoe** [comment]
Hi @enkelmedia, I can proudly tell you that is has been fixed in 15.1, as mentioned in one of your other issues the main workspaces now have discard changes implemented and part of tht journey fixed this issue.
Thanks

**Comment by enkelmedia** [comment]
That’s awesome! 👏🏻 I’ll have to give it a try!
